src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter
+            .ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        Employee[] employees = Arrays.stream(names)
+                .map(Employee::new)
+                .toArray(Employee[]::new);
+
+        for (String salaryData : data) {
+            String[] splittedData = salaryData.split(" ");
+            final String name = splittedData[1];
+            for (Employee employee : employees) {
+                if (employee.getName().equals(name)) {
+                    String dateByData = splittedData[0];
+                    if (checkDate(dateByData, dateFrom, dateTo)) {
```

Comment:
```suggestion
                if (employee.getName().equals(name) && checkDate(splittedData[0], dateFrom, dateTo)) {
```
we can combine 2 ifs